src/prefect/client/client.py

```diff
@@ -567,6 +567,35 @@ def set_flow_run_state(
 
         self.graphql(mutation, state=serialized_state)  # type: Any
 
+    def get_latest_cached_states(
+        self, task_id: str, created_after: datetime.datetime
+    ) -> List["prefect.engine.state.State"]:
+        """
+        Pulls all Cached states for the given task which were created after the provided date.
+
+        Args:
+            - task_id (str): the task id for this task run
+            - created_after (datetime.datetime): the earliest date the state should have been created at
+
+        Returns:
+            - List[State]: a list of Cached states created after the given date
+        """
+        where_clause = {
+            "where": {
+                "state": {"_eq": "Cached"},
+                "task_id": {"_eq": task_id},
+                "created": {"_gte": created_after.isoformat()},
```

Comment:
Sorry I didn't catch this last night -- this field "created" is when the `task_run` itself was created. You'll want to check the `state_timestamp` field instead, which is the time the current state was set for the task_run. 

As an example, a task run might be created today in anticipation of a scheduled flow run, but that run might not start for a month. In a month's time, the task might enter a cached state.

Reply:
That makes too much sense -- good catch and sorry for the oversight!